src/utils/api/btc/transactions.js

```diff
@@ -1,89 +1,83 @@
-import * as bitcoin from 'bitcoinjs-lib';
-import * as popsicle from 'popsicle';
+import bitcoin from 'bitcoinjs-lib';
 import getBtcConfig from './config';
 import { extractAddress, getDerivedPathFromPassphrase } from './account';
 import { validateAddress } from '../../validators';
 import { tokenMap } from '../../../constants/tokens';
+import { getAPIClient } from './network';
 
 /**
  * Normalizes transaction data retrieved from Blockchain.info API
  * @param {Object} data
  * @param {String} data.address Base address to use for formatting transactions
  * @param {Array} data.list Transaction list retrieved from API
  * @param {Number} data.blockHeight Latest block height for calculating confirmation count
- * @param {Number} [netCode=1] Network code of mainnet or testnet. Defaults to testnet (1)
  */
 const normalizeTransactionsResponse = ({
   address,
   list,
-// eslint-disable-next-line max-statements
-}, netCode = 1) => list.map((tx, feeSatoshi, confirmations, timestamp) => {
+  // eslint-disable-next-line max-statements
+}) => list.map(({
+  tx, feeSatoshi, confirmations, timestamp,
+}) => {
   const data = {
     id: tx.txid,
-    timestamp: Number(timestamp) * 1000,
-    confirmations,
+    timestamp: timestamp ? Number(timestamp) * 1000 : null,
+    confirmations: confirmations || 0,
     type: 0,
     data: '',
+    fee: feeSatoshi,
+    token: tokenMap.BTC.key,
   };
 
-  data.fee = feeSatoshi;
-
   const ownedInput = tx.inputs.find(i => i.txDetail.scriptPubKey.addresses.includes(address));
 
   if (ownedInput) {
-    data.senderAddress = address;
+    data.senderId = address;
     const extractedAddress = tx.outputs[0].scriptPubKey.addresses[0];
-    data.recipientAddress = validateAddress(tokenMap.BTC.key, extractedAddress, netCode) === 0 ? extractedAddress : 'Unparsed Address';
+    data.recipientId = validateAddress(tokenMap.BTC.key, extractedAddress) === 0 ? extractedAddress : 'Unparsed Address';
     data.amount = tx.outputs[0].satoshi;
   } else {
+    address = address || tx.inputs[0].txDetail.scriptPubKey.addresses[0];
     const output = tx.outputs.find(o => o.scriptPubKey.addresses.includes(address));
     const extractedAddress = tx.inputs[0].txDetail.scriptPubKey.addresses[0];
-    data.senderAddress = validateAddress(tokenMap.BTC.key, extractedAddress, netCode) === 0 ? extractedAddress : 'Unparsed Address';
-    data.recipientAddress = address;
+    data.senderId = validateAddress(tokenMap.BTC.key, extractedAddress) === 0 ? extractedAddress : 'Unparsed Address';
+    data.recipientId = address;
     data.amount = output.satoshi;
   }
 
   return data;
 });
 
-export const get = ({
-  id,
+export const getTransactions = ({
```

Comment:
I don't see any needs for a separated `getTransactions` and `getSingleTransaction`. why did you implement them instead of a generic get method?

Reply:
We talked about it with @yasharAyari and concluded that a generic get method works in Lisk mobile but in Hub there is more logic for filtering transactions, so it's cleaner to keep them separate. 

There will be a refactor needed to filtering transactions https://github.com/LiskHQ/lisk-hub/issues/1988